oss_src/lambda/pylambda_worker.cpp

```diff
@@ -42,94 +44,137 @@ void memory_release_loop() {
+        if(python_gil_thread_state != nullptr) {
+          PyEval_RestoreThread(python_gil_thread_state);
+          python_gil_thread_state = nullptr;
+        }
+      });
+    
+    try {
+      if(debug_mode) {
+        std::cerr << "INFO: Attempting to initialize python." << std::endl;
+      }
+    
+      graphlab::lambda::init_python(root_path, debug_mode);
+    
+      if(debug_mode) {
+        std::cerr << "INFO: Python initialized successfully." << std::endl;
+      }
+    
+    } catch (const std::string& error) {
+
+      logstream(LOG_ERROR) << "Failed to initialize python (internal exception): " << error << std::endl;
+    
+      if(debug_mode)
+        std::cerr << "ERROR: (internal exception) Failed to initialize python: " << error << std::endl;
+    
+      return 101;
+    } catch (const std::exception& e) {
+
+      if(debug_mode)
+        std::cerr << "ERROR: Failed to initialize python: " << e.what() << std::endl;
+
+      logstream(LOG_ERROR) << "Failed to initialize python: " << e.what() << std::endl;
+    
+      return 102;
+    }
+
+    if(debug_mode) {
+      std::cerr << "INFO: No valid server address, exiting. \n"
+                << "   Example: ipc:///tmp/pylambda_worker\n"
+                << "   Example: tcp://127.0.0.1:10020\n"
+                << "   Example: tcp://*:10020\n"
+                << "   Example: tcp://127.0.0.1:10020 tcp://127.0.0.1:10021\n"
+                << "   Example: ipc:///tmp/unity_test_server --auth_token=secretkey\n"
+                << "   Example: ipc:///tmp/unity_test_server ipc:///tmp/unity_status secretkey\n"
+                << std::endl;
+      return 1; 
+    }
+
+    // Now, release the gil and continue. 
+    python_gil_thread_state = PyEval_SaveThread();
+    
+    graphlab::shmipc::server shm_comm_server;
+    bool has_shm = shm_comm_server.bind();
+    // construct the server
+    cppipc::comm_server server(std::vector<std::string>(), "", server_address);
+
+    server.register_type<graphlab::lambda::lambda_evaluator_interface>([&](){
+        if (has_shm) {
+          return new graphlab::lambda::pylambda_evaluator(&shm_comm_server);
+        } else {
+          return new graphlab::lambda::pylambda_evaluator();
+        }
+      });
+    server.register_type<graphlab::lambda::graph_lambda_evaluator_interface>([](){
+        return new graphlab::lambda::graph_pylambda_evaluator();
+      });
+
+    server.start();
 
 #ifdef HAS_TCMALLOC
-  graphlab::thread memory_release_thread;
-  memory_release_thread.launch(memory_release_loop);
+    graphlab::thread memory_release_thread;
+    memory_release_thread.launch(memory_release_loop);
 #endif
 
-  wait_for_parent_exit(parent_pid);
+    wait_for_parent_exit(parent_pid);
 
 #ifdef HAS_TCMALLOC
-  stop_memory_release_thread = true;
-  memory_release_cond.signal();
-  memory_release_thread.join();
+    stop_memory_release_thread = true;
+    memory_release_cond.signal();
+    memory_release_thread.join();
 #endif
```

Comment:
We can delete these all together. In fact, we should delete these. TCMalloc will generally not be loaded since Python is the driver.